tron/mesos.py

```diff
@@ -32,17 +31,33 @@ def get_mesos_leader(master_address):
     return '{}:{}'.format(urlparse(response.url).hostname, MESOS_MASTER_PORT)
 
 
-def get_mesos_cluster(master_address):
-    global frameworks
-    if master_address not in frameworks:
-        frameworks[master_address] = MesosCluster(master_address)
-    return frameworks[master_address]
+class ClusterRepository:
+    """A class that stores MesosCluster objects and configuration."""
 
+    clusters = {}
+    mesos_enabled = False
 
-def shutdown_frameworks():
-    global frameworks
-    for name, framework in frameworks.items():
-        framework.stop()
+    def __init__(self):
+        super(ClusterRepository, self).__init__()
+
+    @classmethod
+    def get_cluster(cls, master_address):
+        if master_address not in cls.clusters:
+            cls.clusters[master_address] = MesosCluster(
```

Comment:
if leader election results in different master address, will we have multiple frameworks in single cluster? we should probably index by cluster name?

Reply:
The master address is the fqdn, which points to all of the masters. It's like "mesos-norcal-prod.yelp" which is basically the cluster name.